phylogeny/prototypeSelection.py

```diff
@@ -201,3 +202,173 @@ def prototype_selection_constructive_maxdist(dm: DistanceMatrix,
 
     # return the ids of the selected prototype elements
     return [dm.ids[idx] for idx, x in enumerate(uncovered) if not x]
+
+
+def _protoclass(dm, epsilon):
+    '''Heuristically select n prototypes for a fixed epsilon radius.
+
+       A ball is drawn around every element in the distance matrix with radius
+       epsilon. The element whoes ball covers most other elements is selected
+       as prototype. All such covered elements and the new prototype are
+       removed for the next round. This is repeated until no balls cover more
+       than its center element.
+       This idea is adapted from [1] with the difference that we only deal with
+       a single class.
+
+    Parameters
+    ----------
+    dm: skbio.stats.distance.DistanceMatrix
+        Pairwise distances for all elements in the full set S.
+    epsilon: float
+        Radius for the balls to be "drawn". As a rule of thumb, the larger
+        epsilon, the less prototypes are found.
+
+    Returns
+    -------
+    list of str
+        A sequence holding selected prototypes, i.e. a sub-set of the
+        elements in the distance matrix.
+
+    Notes
+    -----
+    function signature with type annotation for future use with python >= 3.5:
+    def _protoclass(dm: DistanceMatrix, epsilon: float) -> List[str]:
+
+    [1] Jacob Bien and Robert Tibshirani.
+        "Prototype selection for interpretable classification."
+        The Annals of Applied Statistics (2011): 2403-2424.
+    '''
+
+    # is an element (column) covered by the epsilon ball of another element
+    # (row)
+    B = dm.data < epsilon
+    # tracks which elements are covered by prototypes
+    covered = [0] * dm.shape[0]
```

Comment:
I like @sjanssen2 's original solution, because it is a more general solution. One can do `['whatever'] * n`.